validator/accounts/status.go

```diff
@@ -0,0 +1,240 @@
+package accounts
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/plugin/ocgrpc"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+)
+
+// ValidatorStatusMetadata holds all status information about a validator.
+type ValidatorStatusMetadata struct {
+	PublicKey []byte
+	Metadata  *ethpb.ValidatorStatusResponse
+}
+
+// MaxRequestLimit specifies the max concurrent grpc requests allowed
+// to fetch account statuses.
+const MaxRequestLimit = 5 // XXX: Should create flag to make parameter configurable.
+
+// MaxRequestKeys specifies the max amount of public keys allowed
+// in a single grpc request, when fetching account statuses.
+const MaxRequestKeys = 2000 // XXX: This is an arbitrary number. Used to limit time complexity
+// of sorting a single batch of status requests. Do not make this number too big.
+
+// RunStatusCommand is the entry point to the `validator status` command.
+func RunStatusCommand(
+	pubkeys [][]byte,
+	withCert string,
+	endpoint string,
+	maxCallRecvMsgSize int,
+	grpcRetries uint,
+	grpcHeaders []string) error {
+	dialOpts, err := constructDialOptions(maxCallRecvMsgSize, withCert, grpcHeaders, grpcRetries)
```

Comment:
Can you not construct the GRPC client in this command, this should be done on the outside of the method.